src/Rules/ServerRuleMetadataProvider.cs

```diff
@@ -61,30 +61,32 @@ public ServerRuleMetadataProvider(ISonarQubeService service, ILogger logger)
 
         public async Task<IRuleInfo> GetRuleInfoAsync(SonarCompositeRuleId ruleId, string qualityProfileKey, CancellationToken token)
         {
-            SonarQubeRule sqRule = null;
             try
             {
-                sqRule = await service.GetRuleByKeyAsync(ruleId.ToString(), qualityProfileKey, token);
+                var sqRule = await service.GetRuleByKeyAsync(ruleId.ToString(), qualityProfileKey, token);
+
+                if (sqRule == null) { return null; }
+
+                var descriptionSections = sqRule.DescriptionSections?.Select(ds => ds.ToDescriptionSection()).ToList();
+
+                return new RuleInfo(sqRule.RepositoryKey,
+                    sqRule.GetCompositeKey(),
+                    HtmlXmlCompatibilityHelper.EnsureHtmlIsXml(sqRule.Description),
+                    sqRule.Name,
+                    sqRule.Severity.ToRuleIssueSeverity(),
+                    sqRule.IssueType.ToRuleIssueType(),
+                    sqRule.IsActive,
+                    sqRule.Tags,
+                    descriptionSections,
+                    sqRule.EducationPrinciples,
+                    HtmlXmlCompatibilityHelper.EnsureHtmlIsXml(sqRule.HtmlNote));
             }
             catch (Exception ex)
             {
                 logger.WriteLine(Resources.ServerMetadataProvider_GetRulesError, ruleId.ToString(), ex.Message);
             }
-            if (sqRule == null) return null;
-
-            var descriptionSections = sqRule.DescriptionSections.Select(ds => ds.ToDescriptionSection()).ToList();
 
-            return new RuleInfo(sqRule.RepositoryKey,
-                sqRule.GetCompositeKey(),
-                HtmlXmlCompatibilityHelper.EnsureHtmlIsXml(sqRule.Description),
-                sqRule.Name,
-                sqRule.Severity.ToRuleIssueSeverity(),
-                sqRule.IssueType.ToRuleIssueType(),
-                sqRule.IsActive,
-                sqRule.Tags,
-                descriptionSections,
-                sqRule.EducationPrinciples,
-                HtmlXmlCompatibilityHelper.EnsureHtmlIsXml(sqRule.HtmlNote));
```

Comment:
I moved the rest of this conversion code inside the exception handler. The new behaviour is that null will be returned in the event of any exceptions when trying to fetch the rule info from the server.

Previously, an exception was being thrown in line 75 of this class. It was outside the exception handler, so wasn't caught in this method. In fact, it wasn't caught anywhere. Fortunately, it was happening on a background thread, so VS was silently swallowing it.

The impact was that if the user clicked to see error help for a secret in Connected Mode, nothing would happen from the user's point of view (the exception would be thrown and silently squashed).

The new behaviour is that the exception will be caught and logged. This service will return `null`, and the education service that calls it will fall back on displaying the local help, if it exists.

Testing: there is already a unit test that checks that exceptions from the SonarQube service are caught and logged. Unfortunately, there aren't any injection points in the product code after the service call, so we can't unit test the new behaviour. However, I have manually stepped through it when debugging.